crates/aptos-api-tester/src/main.rs

```diff
@@ -0,0 +1,578 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use anyhow::{ensure, Context, Result};
+use aptos_api_types::{Block, HashValue, U64};
+use aptos_rest_client::{Account, Client, FaucetClient};
+use aptos_sdk::coin_client::CoinClient;
+use aptos_sdk::types::LocalAccount;
+use aptos_types::account_address::AccountAddress;
+use aptos_types::transaction::authenticator::AuthenticationKey;
+use once_cell::sync::Lazy;
+use rand::Rng;
+use std::str::FromStr;
+use url::Url;
+
+// global parameters (todo: make into clap)
+static HAS_FAUCET_ACCESS: bool = false;
+
+// network urls
+static DEVNET_NODE_URL: Lazy<Url> =
+    Lazy::new(|| Url::parse("https://fullnode.devnet.aptoslabs.com").unwrap());
+static DEVNET_FAUCET_URL: Lazy<Url> =
+    Lazy::new(|| Url::parse("https://faucet.devnet.aptoslabs.com").unwrap());
+static TESTNET_NODE_URL: Lazy<Url> =
+    Lazy::new(|| Url::parse("https://fullnode.testnet.aptoslabs.com").unwrap());
+
+// static accounts to use
+// don't send coins to TEST_ACCOUNT_1
+static TEST_ACCOUNT_1: Lazy<AccountAddress> = Lazy::new(|| {
+    AccountAddress::from_hex_literal(
+        "0x7fed760c508a8885e1438c90f81dea6940ad53b05527fadddd3851f53342d7c4",
+    )
+    .unwrap()
+});
+static TEST_ACCOUNT_2: Lazy<AccountAddress> = Lazy::new(|| {
+    AccountAddress::from_hex_literal(
+        "0xf6cee5c359839a321a08af6b8cfe4a6e439db46f4942e21b6845aa57d770efdb",
+    )
+    .unwrap()
+});
+
+// return values
+static SKIP_NO_FAUCET_ACCESS: &str = "This test requires faucet access.";
```

Comment:
This actually brings up a good point that maybe just "success" and "failure" are not the only two outputs if you were to keep track of it, perhaps "skip" is another one if the test doesn't apply.  An enum can help with this on the Error, though it makes it a little less convenient then just calling `ensure!()`